src/Tracking/TrackerSnapshot.php

```diff
@@ -77,18 +79,25 @@ protected function get_settings(): array {
 		/** @var TargetAudience $target_audience */
 		$target_audience = $this->container->get( TargetAudience::class );
 		$mc_settings     = $this->options->get( OptionsInterface::MERCHANT_CENTER );
+		/** @var AdsService $ads_service */
+		$ads_service = $this->container->get( AdsService::class );
+		/** @var SetupCampaignNote $campaign_setup */
+		$campaign_setup = $this->container->get( SetupCampaignNote::class );
```

Comment:
As mentioned in the previous comment we want to get data from the AdsService and the MerchantCenterService, not directly from a Note which has a different function.